drivers/dma/dma_andes_atcdmac300.c

```diff
@@ -0,0 +1,435 @@
+/*
+ * Copyright (c) 2023 Andes Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <errno.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/dma.h>
+#include <soc.h>
+#include "dma_andes_atcdmac300.h"
+
+#define DT_DRV_COMPAT andestech_atcdmac300
+
+#define LOG_LEVEL CONFIG_DMA_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(dma_andes_atcdmac300);
+
+K_HEAP_DEFINE(dma_chain, 1024);
```

Comment:
a pool of chain objects would be more sensible here than a heap and simpler, generally Zephyr drivers shouldn't be using a general purpose heap given the misra rules

Reply:
I got it. We weren't aware of that.
We have update the source code.